File: AI/Java/AgentTest/demo2/src/main/java/app/tools/Shop.java
package app.tools;

import app.models.Inventory;
import app.models.ItemInfo;
import dev.langchain4j.agent.tool.P;
import dev.langchain4j.agent.tool.Tool;

public class Shop {
    
    private final Inventory model = new Inventory();

    @Tool("Gets unit-price and stock-level of an item.")
    public ItemInfo fetchItem(@P("Singularized name of item.") String item) {
        return model.store().get(item.toLowerCase());
    }

    @Tool("Gets percentage discount offered on bulk purchase.")
    public static float offerDiscount(@P("Number of items.") int quantity) {
        if(quantity >= 20)
            return 7.5f;
        if(quantity >= 10)
            return 5.0f;
        return 0;
    }
}

File: AI/DotNet/AgentTest/DemoApp2/Tools/Shop.cs
using System.ComponentModel;
using DemoApp.Models;

namespace DemoApp.Tools;

public class Shop
{
    private readonly Inventory model = new ();

    [Description("Gets unit-price and stock-level of an item.")]
    public ItemInfo? FetchItem([Description("Singularized name of the item.")] string item)
    {
        return model.Store.GetValueOrDefault(item.ToLower());
    }

    [Description("Gets percentage discount for bulk purchase.")]
    public static float OfferDiscount([Description("Number of items")] int quantity)
    {
        if(quantity >= 20)
            return 7.5f;
        if(quantity >= 10)
            return 5.0f;
        return 0;
    }
}